store/sqlstore/post_persistent_notification_store.go

```diff
@@ -0,0 +1,251 @@
+	for i := 0; i < count; i += paginationLimit {
+		j := utils.MinInt(i+paginationLimit, count)
+
+		builder := s.getQueryBuilder().
+			Update("PersistentNotifications").
+			Set("LastSentAt", lastSentAt).
+			Set("SentCount", sq.Expr("SentCount+1")).
+			Where(sq.Eq{"PostId": postIds[i:j]})
+
+		_, err := s.GetMasterX().ExecBuilder(builder)
+		if err != nil {
+			return errors.Wrapf(err, "failed to update lastSentAt for posts %s", postIds[i:j])
+		}
+	}
+
+	return nil
+}
+
+// DeleteByChannel in batches of 1000
+func (s *SqlPostPersistentNotificationStore) DeleteByChannel(channelIds []string) error {
+	count := len(channelIds)
+	if count == 0 {
+		return nil
+	}
+
+	deleteAt := model.GetMillis()
+	for i := 0; i < count; i += paginationLimit {
+		j := utils.MinInt(i+paginationLimit, count)
+
+		var builder sq.UpdateBuilder
+		builderType := s.getQueryBuilder()
+		if s.DriverName() == model.DatabaseDriverMysql {
+			builder = builderType.
+				Update("PersistentNotifications, Posts").
+				Set("PersistentNotifications.DeleteAt", deleteAt)
+		}
+
+		if s.DriverName() == model.DatabaseDriverPostgres {
+			builder = builderType.
+				Update("PersistentNotifications").
+				Set("DeleteAt", deleteAt).
+				From("Posts")
+		}
+
+		builder = builder.Where(sq.And{
+			sq.Expr("Id = PostId"),
+			sq.Eq{"ChannelId": channelIds[i:j]},
+		})
+
+		_, err := s.GetMasterX().ExecBuilder(builder)
+		if err != nil {
+			return errors.Wrapf(err, "failed to delete notifications for channels %s", channelIds[i:j])
+		}
+	}
+
+	return nil
+}
+
+// DeleteByTeam in batches of 1000
+func (s *SqlPostPersistentNotificationStore) DeleteByTeam(teamIds []string) error {
+	count := len(teamIds)
+	if count == 0 {
+		return nil
+	}
+
+	deleteAt := model.GetMillis()
+	for i := 0; i < count; i += paginationLimit {
+		j := utils.MinInt(i+paginationLimit, count)
+
+		var builder sq.UpdateBuilder
+		builderType := s.getQueryBuilder()
+		if s.DriverName() == model.DatabaseDriverMysql {
+			builder = builderType.
+				Update("PersistentNotifications, Posts, Channels").
+				Set("PersistentNotifications.DeleteAt", deleteAt)
+		}
+
+		if s.DriverName() == model.DatabaseDriverPostgres {
+			builder = builderType.
+				Update("PersistentNotifications").
+				Set("DeleteAt", deleteAt).
+				From("Posts, Channels")
+		}
+
+		builder = builder.Where(sq.And{
+			sq.Expr("Posts.Id = PersistentNotifications.PostId"),
+			sq.Expr("Posts.ChannelId = Channels.Id"),
+			sq.Eq{"Channels.TeamId": teamIds[i:j]},
+		})
+
+		_, err := s.GetMasterX().ExecBuilder(builder)
+		if err != nil {
+			return errors.Wrapf(err, "failed to delete notifications for teams %s", teamIds[i:j])
+		}
+	}
+
+	return nil
+}
+
+func getPersistentNotificationsPaginationBuilder(builder sq.SelectBuilder, pagination model.CursorPagination) sq.SelectBuilder {
+	var sort string
+	if pagination.Direction != "" {
+		if pagination.Direction == "up" {
+			sort = "DESC"
+		} else if pagination.Direction == "down" {
+			sort = "ASC"
+		}
```

Comment:
I think it could be because, we may not want to _directly_ bind the user-input(strings) into the sql-query. If it were asc/desc, developers will likely just use the `pagination.Direction` (which could contain anything) into the query.

Or maybe it was just random. 👀

Reply:
You're right, it does discourage interpolation directly into the query string. But man, as a user of the API, mapping `up` to "descending" is weird :p

Reply:
yeah that is weird 😅. I'll update it.